src/PeoplePage.js

```diff
@@ -0,0 +1,73 @@
+import React, { useState } from 'react';
+import { useHistory, useLocation } from 'react-router-dom';
+import peopleFromServer from './people';
+import PeopleFilter from './PeopleFilter';
+import PeopleTable from './PeopleTable';
+import { debounce } from './debounce';
+
+const preparedPeople = peopleFromServer.map(
+  (person, index) => (
+    {
+      ...person,
+      id: index + 1,
+      age: person.died - person.born,
+      century: Math.ceil(person.died / 100),
+    }
+  )
+);
+
+const PeoplePage = () => {
+  const [people] = useState([...preparedPeople]);
+
+  const location = useLocation();
+  const searchParams = new URLSearchParams(location.search);
+
+  const history = useHistory();
+
+  const setQuery = debounce((input) => {
+    searchParams.set('query', input);
+    history.push({ search: searchParams.toString() });
+  }, 500);
+
+  const setSortBy = (sortBy) => {
+    searchParams.set('sortBy', sortBy);
+    history.push({ search: searchParams.toString() });
+  };
+
+  const getSortedPeople = () => {
+    const sortBy = searchParams.get('sortBy');
+
+    switch (typeof people[0][sortBy]) {
+      case 'string':
+        return (people
```

Comment:
```suggestion
        return ([...people]
```
Сортировка массива происходит на месте
Учитывая что эта переменная объявлена у тебя через useState - желательно менять ее через функцию setPeople из этого же useState